guides/source/active_record_migrations.md

```diff
@@ -927,43 +1172,66 @@ number associated with the previous migration you can run:
 $ bin/rails db:rollback
 ```
 
-This will rollback the latest migration, either by reverting the `change`
-method or by running the `down` method. If you need to undo
-several migrations you can provide a `STEP` parameter:
+This will rollback the latest migration, either by reverting the `change` method
+or by running the `down` method. If you need to undo several migrations you can
+provide a `STEP` parameter:
 
 ```bash
 $ bin/rails db:rollback STEP=3
 ```
 
 The last 3 migrations will be reverted.
 
-The `db:migrate:redo` command is a shortcut for doing a rollback and then migrating
-back up again. As with the `db:rollback` command, you can use the `STEP` parameter
-if you need to go more than one version back, for example:
+In some cases where you modify a local migration and would like to rollback that
+specific migration before migrating back up again, you can use the
+`db:migrate:redo` command. As with the `db:rollback` command, you can use the
+`STEP` parameter if you need to go more than one version back, for example:
 
 ```bash
 $ bin/rails db:migrate:redo STEP=3
 ```
 
-Neither of these rails commands do anything you could not do with `db:migrate`. They
-are there for convenience, since you do not need to explicitly specify the
-version to migrate to.
+NOTE: You could get the same result using `db:migrate`. However, these are there
+for convenience so that you do not need to explicitly specify the version to
+migrate to.
+
+#### Transactions
+
+In databases that support transactions with statements that change the schema,
+each migration is wrapped in a transaction.
+
+INFO: A transaction ensures that if a migration fails partway through, any
+changes that were successfully applied are rolled back, maintaining database
+consistency. This means that either all operations within the transaction are
+executed successfully, or none of them are, preventing the database from being
+left in an inconsistent state if an error occurs during the transaction.
+
+If the database does not support transactions with statements that change the
+schema, then when a migration fails, the parts of it that have succeeded will
+not be rolled back. You will have to rollback the changes manually.
 
-### Setup the Database
+NOTE: There are certain queries that can't run inside a transaction. If your
+adapter supports DDL transactions you can use `disable_ddl_transaction!` method
+to disable transactions for a single migration, allowing these queries to be
+executed outside the transactional boundary.
```

Comment:
In a pedantic sense, this isn't strictly true. Even for concurrently in Postgres there's a transaction that initially commits the index definition in an invalid state. There are actually three total transactions!

I can contribute a writeup on this if it helps. It would be specific to Postgres though and is likely too much scope/too detailed for this section, but possibly as an appendix.